slack/models.py

```diff
@@ -0,0 +1,142 @@
+import datetime
+
+from django.db import models, connection
+
+"""
+`ts` fields from slack are actually IDs that can be interpreted into
+timestamps.  We store them in a string as we get them from slack instead
+of a DateTimeField to be able to pass them back exactly to slack without
+precision or something causing round-tripping errors.
+"""
+
+
+def parse_ts(ts):
+    seconds = float(ts)
+    return datetime.datetime.fromtimestamp(seconds, tz=datetime.timezone.utc)
+
+
+class SlackUser(models.Model):
+    id = models.CharField(max_length=16, primary_key=True)
```

Comment:
Are you sure we shouldn't store these as `slack_id` and let Postgres store an `AutoField` as the actual db id? "Natural" primary keys are generally best avoided. Maybe in this case it's ok since these models really don't make sense without their unique slack identifier, though.

Reply:
A surrogate key wouldn't be used anywhere, so it doesn't need to exist. I think it would just cause confusion between our arbitrary ID and the Slack ID.